cmake/TivaCMake/FindCodeComposerStudio.cmake

```diff
@@ -36,28 +42,59 @@ The following cache variables may also be set:
 if(NOT CodeComposerStudio_FOUND)
   # Gather directory names to search
   file(GLOB CodeComposerStudio_ROOTS
-    $ENV{HOME}/ti/ccs* $ENV{HOME}/ccs* /opt/ccs* /opt/ti/ccs*)
+    $ENV{HOME}/ti/ccs* $ENV{HOME}/ccs* /opt/ccs* /opt/ti/ccs* /Applications/ti/ccs* )
+
 
   # Find the code composer studio executable
-  find_program(CodeComposerStudio_EXECUTABLE
-    NAMES ccstudio
+ 
+  find_path(CodeComposerStudio_ROOT_DIR
+NAMES ccs  
     HINTS ${CodeComposerStudio_ROOTS}
-    PATH_SUFFIXES ccs/eclipse
-    NO_DEFAULT_PATH
+ 
+ 
+)
+
+  if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+	  find_program(CodeComposerStudio_EXECUTABLE
+	    NAMES ccstudio
+	    HINTS ${CodeComposerStudio_ROOTS}
+	    PATH_SUFFIXES ccs/eclipse
+	    NO_DEFAULT_PATH
     )
 
+  	get_filename_component(CodeComposerStudio_EXEPATH ${CodeComposerStudio_EXECUTABLE} DIRECTORY)
+  	get_filename_component(CodeComposerStudio_ROOT_DIR ${CodeComposerStudio_EXEPATH}/../.. ABSOLUTE)
+  	# We want ROOT_DIR to be in the cache and also be of type PATH
+  	set(CodeComposerStudio_ROOT_DIR ${CodeComposerStudio_ROOT_DIR} CACHE PATH "Base Directory for Code Composer Studio")
+
+
+   else()
+	set(CodeComposerStudio_ROOT_DIR ${CodeComposerStudio_BASE} CACHE PATH "Base Directory for Code Composer Studio")
```

Comment:
CodeComposerStudio_BASE is not defined anywhere else, so I'm not sure how this line works...